metaflow/plugins/kfp/kfp.py

```diff
@@ -643,6 +672,17 @@ def _set_container_resources(
             container_op.add_affinity(affinity)
             container_op.add_toleration(toleration)
 
+        if (
+            not kfp_component.accelerator_decorator
+            and "gpu" not in resource_requirements
```

Comment:
this should probably be an `else` case for `if kfp_component.accelerator_decorator:`